lib/ipmi_user.c

```diff
@@ -477,16 +481,47 @@ ipmi_user_build_password_prompt(uint8_t user_id)
  *
  * @returns pointer to char with password
  */
-char *
-ask_password(uint8_t user_id)
+static int
+ask_password(uint8_t user_id, char *buf, size_t buflen)
 {
-	const char *password_prompt =
-		ipmi_user_build_password_prompt(user_id);
-# ifdef HAVE_GETPASSPHRASE
-	return getpassphrase(password_prompt);
-# else
-	return (char*)getpass(password_prompt);
-# endif
+	const char *password_prompt;
+	char *sbuf;
+	size_t len;
+	int ret = -1;
+	char* (*password_routine)(const char *prompt);
+
+	/*
+	 * Note: getpass appears to be a trivial portable password routine,
+	 * however it is considered obsolete by most modern OSs. Unfortunately
+	 * there is not really a portable solution that was obvious so getpass
+	 * is currently to remain in use.
+	 */
+#ifdef HAVE_GETPASSPHRASE
+	password_routine = &getpassphrase;
+#else
+	password_routine = &getpass;
+#endif
+
+	password_prompt = ipmi_user_build_password_prompt(user_id);
+	if (buf && buflen > 0) {
+		sbuf = password_routine(password_prompt);
```

Comment:
nit: you can reduce the indentation with some early bails - I don't know what the policy is on these, but if sbuf == NULL) { log; return ret; } etc etc